network-libp2p/src/connection_pool/behaviour.rs

```diff
@@ -364,6 +364,31 @@ impl ConnectionPoolBehaviour {
             .choose_multiple(&mut thread_rng(), num_peers)
     }
 
+    pub fn choose_peers_to_dial_by_services(&self, services: Services) -> Vec<PeerId> {
+        let num_peers = usize::min(
+            self.config.peer_count_desired - self.peer_ids.num_connected(),
+            self.config.dialing_count_max - self.peer_ids.num_dialing(),
+        );
```

Comment:
I think this is not right. This in most cases would lead to zero peers since we might have reached the _desired_ number of connections (which is what this `min` looks for)